pkg/cloudprovider/cloudprovider.go

```diff
@@ -228,7 +228,7 @@ func (c *CloudProvider) isAMIDrifted(ctx context.Context, machine *v1alpha5.Mach
 		return false, err
 	}
 	instance, err := c.instanceProvider.Get(ctx, instanceID)
-	if err != nil {
+	if cloudprovider.IgnoreMachineNotFoundError(err) != nil {
```

Comment:
Won't this cause a problem on line 234, since we'll try to dereference *instance.ImageId